libs/api/protocol/src/lib/infra/view-model/unit-input.view-model.ts

```diff
@@ -0,0 +1,40 @@
+import { Field, InputType, registerEnumType } from '@nestjs/graphql';
+import {
+	IsEnum,
+	IsMongoId,
+	IsNotEmpty,
+	IsString,
+	MaxLength,
+	ValidateIf,
+} from 'class-validator';
+
+import { Validatable } from '@kordis/api/shared';
+
+export enum UnitInputType {
+	REGISTERED_UNIT = 'REGISTERED_UNIT',
+	UNKNOWN_UNIT = 'UNKNOWN_UNIT',
+}
+
+registerEnumType(UnitInputType, {
+	name: 'UnitInputType',
+});
+
+@InputType()
+export class UnitInput extends Validatable {
```

Comment:
we should validate, that only the id OR the name is set, depending the UnitInputType. Cant you have a union input type that would take care of this?